### PR TITLE
Made implicit conversions to/from Raster and ProjectedRaster deprecated.

### PR DESCRIPTION
**Pull request by metasim**
Partially addresses #2829.

### CODE REVIEW

File: raster/src/main/scala/geotrellis/raster/Raster.scala [hunk 1]
```diff
@@ -33,25 +30,29 @@ object Raster {
...
     * Implicit conversion from a [[Raster]] to a PolygonFeature.
     */
+  @deprecated("Implicit conversions considered unsafe", "2.1.1")
   implicit def rasterToFeature[T <: CellGrid](r: Raster[T]): PolygonFeature[T] =
     r.asFeature
 
   /**
     * Implicit conversion from a PolygonFeature to a [[Raster]].
     */
+  @deprecated("Implicit conversions considered unsafe", "2.1.1")
```
[the remark this answers is not in the document]
Comment (metasim) [reply raster/src/main/scala/geotrellis/raster/Raster.scala]: When you say "remove them" do you mean the `@deprecation` annotations, or the messages?

File: doc-examples/src/main/scala/geotrellis/doc/examples/raster/MatchingRasters.scala [hunk 2]
```diff
@@ -38,9 +35,11 @@ object MatchingRasters {
     val cropped2 =
       raster2
         .reproject(raster1.crs).raster
-        .resample(raster1.rasterExtent)
+        .resample(raster1.raster.rasterExtent)
         .crop(areaOfInterest)
 
+    //.resample(ProjectedRaster.projectedToRaster(raster1).rasterExtent))(withTileMethods())
```
[the remark this answers is not in the document]
Comment (metasim) [reply doc-examples/src/main/scala/geotrellis/doc/examples/raster/MatchingRasters.scala]: Yep

**Comment by metasim** [comment]
NB: Build failed only due to one config timing out.

File: raster/src/main/scala/geotrellis/raster/Raster.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (pomadchin) [reply raster/src/main/scala/geotrellis/raster/Raster.scala]: Conversions, we can just mention in docs that we removed these implicits.

[the remark this answers is not in the document]
Comment (echeipesh) [reply raster/src/main/scala/geotrellis/raster/Raster.scala]: Back-porting it and publishing the deprecation warnings with `2.1.1` is defiantly a good-guy move and we should do it.

[the remark this answers is not in the document]
Comment (metasim) [reply raster/src/main/scala/geotrellis/raster/Raster.scala]: I'm still confused as to what you want me to do with this part.

**Pull request merged**